Sources/NIO/EventLoop.swift

```diff
@@ -92,21 +94,45 @@ public final class RepeatedTask {
     ///  This means immediate cancellation is not guaranteed.
     ///
     /// The safest way to cancel is by using the passed reference of `RepeatedTask` inside the task closure.
-    public func cancel() {
+    ///
+    /// If the promise parameter is not `nil`, the passed promise is fulfilled when cancellation is complete.
+    /// Passing a promise does not prevent fulfillment of any promise provided on original task creation.
+    public func cancel(andNotify promise: EventLoopPromise<Void>? = nil) {
         if self.eventLoop.inEventLoop {
-            self.cancel0()
+            self.cancel0(andNotify: promise)
         } else {
             self.eventLoop.execute {
-                self.cancel0()
+                self.cancel0(andNotify: promise)
             }
         }
     }
 
-    private func cancel0() {
+    private func cancel0(andNotify localCancellationPromise: EventLoopPromise<Void>?) {
```

Comment:
Because the default is only there on the public API as a convenience; since this method is a `private` implementation detail, it's better to force callers to be explicit.

Reply:
Ah. Makes sense, thanks 😃